src/main/java/fr/sncf/osrd/infra/railscript/RSExprState.java

```diff
@@ -87,7 +87,7 @@ boolean hasDelaySlotChanged(int index) {
         return argScopeOffset + index == lastUpdatedDelaySlot;
     }
 
-    RSExprState(RSExpr<T> rootExpr, int argSlotCount, int delaySlotCount) {
+    public RSExprState(RSExpr<T> rootExpr, int argSlotCount, int delaySlotCount) {
```

Comment:
Need to add doc to pass checkstyle

```suggestion
    /** Create a new RSExprState */
    public RSExprState(RSExpr<T> rootExpr, int argSlotCount, int delaySlotCount) {
```